### ISSUE TITLE
[Feature Request] Ability to use Node's URLSearchParams

### ISSUE THREAD

**Issue by ryuuji3** [axios/axios#1891]
#### Summary

Currently, Axios requests accept a params object which can be a simple object, or a URLSearchParams object... but only on the front-end.

Prompted by: #1699 

Recently, I ran into this myself and only after re-reading the documentation I realized that I skimmed the part where URLSearchParams was not a NodeJS option.
Unfamiliar with the complications of doing so, I'd like to request that Node's URLSearchParams also be an acceptable option. 

My work-around is to use toString() and append it to the URL, but it would be nice to simply use the params object as it's intended. 
Is there a technical reason we can't also allow it?

Example usage on NodeJS.

```ts
const params = new URLSearchParams({
   Key: "12345",
   Text: "example query"
});

// What I'd like:
return requestInterface.get("/Find", {
  params
};

// What I currently can do:
return requestInterface.get(`/Find?${params.toString()}`);
```

Using URLSearchParams is simply about having static type safety and prevent runtime exceptions for invalid search params... which is awfully nice to have.

**Comment by mindriven** [comment]
I like your idea, I'll try to create a PR. Just give me a some time :-)

**Comment by ryuuji3** [comment]
@mindriven Thank you for jumping on that so fast!

**Comment by jasonsaayman** [comment]
Merged #1900
